libraries/Matter/examples/MatterThermostat/MatterThermostat.ino

```diff
@@ -0,0 +1,248 @@
+// User can set a local temperature using the Serial input (type a number and press Enter)
+// User can change the thermostat mode using the Matter APP (smartphone)
+// The loop will simulate a heating and cooling system and the associated local temperature change
+void loop() {
+  static uint32_t timeCounter = 0;
+
+  // Simulate the heating and cooling systems
+  static bool isHeating = false;
+  static bool isCooling = false;
+
+  // check if a new temperature is typed in the Serial Monitor
+  readSerialForNewTemperature();
+
+  // simulate thermostat with heating/cooling system and the associated local temperature change, every 10s
+  if (!(timeCounter++ % 20)) {  // delaying for 500ms x 20 = 10s
+    float localTemperature = getSimulatedTemperature(isHeating, isCooling);
+    // Print the current thermostat local temperature value
+    Serial.printf("Current Local Temperature is %.01fC\r\n", localTemperature);
+    SimulatedThermostat.setLocalTemperature(localTemperature); // publish the new temperature value
+
+    // Simulate the thermostat control system - User has 4 modes: OFF, HEAT, COOL, AUTO
+    if (SimulatedThermostat.getMode() == MatterThermostat::THERMOSTAT_MODE_OFF) {
+      // turn off the heating and cooling systems
+      isHeating = false;
+      isCooling = false;
+    }
+    // User APP has set the thermostat to AUTO mode -- keeping the tempeature between both setpoints
+    if (SimulatedThermostat.getMode() == MatterThermostat::THERMOSTAT_MODE_AUTO) {
+      // check if the heating system should be turned on or off
+      if (localTemperature < SimulatedThermostat.getHeatingSetpoint() + SimulatedThermostat.getDeadBand()) {
+        // turn on the heating system and turn off the cooling system
+        isHeating = true;
+        isCooling = false;
+      }
+      if (localTemperature > SimulatedThermostat.getCoolingSetpoint() - SimulatedThermostat.getDeadBand()) {
+        // turn off the heating system and turn on the cooling system
+        isHeating = false;
+        isCooling = true;
+      }
+    }
+    // User APP has set the thermostat to AUTO mode -- keeping the tempeature between both setpoints
+    if (SimulatedThermostat.getMode() == MatterThermostat::THERMOSTAT_MODE_AUTO) {
+      // check if the heating system should be turned on or off
+      if (localTemperature < SimulatedThermostat.getHeatingSetpoint() + SimulatedThermostat.getDeadBand()) {
+        // turn on the heating system and turn off the cooling system
+        isHeating = true;
+        isCooling = false;
+      }
+      if (localTemperature > SimulatedThermostat.getCoolingSetpoint() - SimulatedThermostat.getDeadBand()) {
+        // turn off the heating system and turn on the cooling system
+        isHeating = false;
+        isCooling = true;
+      }
+    }
+    // Simulate the heating system - User has turned the heating system ON
+    if (SimulatedThermostat.getMode() == MatterThermostat::THERMOSTAT_MODE_HEAT) {

```

Comment:
`elif`? can't be AUTO, HEAT and COOL at the same time, right?

Reply:
No. Mode is exclusively one of these: OFF, AUTO, HEAT or COOL.
It is about the temperature operation control.
The accessory can be initialized with a Control Sequence that may have a combination of those.
Example: Thermostat that only can be OFF or HEATING, will begin(SEQ_OFF_HEAT) and only can be in modes OFF or HEAT.

Reply:
Changed the code to use a `switch()`